### ISSUE TITLE
val buffer and val of_bigarray implement the same

### ISSUE THREAD

**Issue by reynir** [mirage/ocaml-cstruct#311]
The two functions do exactly the same except for different messages in the `Invalid_argument _` exception. It is not clear from the documentation.

Discovered while answering https://github.com/mirage/ocaml-cstruct/issues/310

**Comment by reynir** [comment]
It seems it was added in https://github.com/mirage/ocaml-cstruct/commit/6e82959d2ea063e9354fafa05cb32fce336e246a#diff-1c794fc18075ec0a83c9f7b2ad768653e0ec3a60d60323e713d2f04d5b3c7c2aR82 while `of_bigarray` is much older.

**Comment by dinosaure** [comment]
Yes, that's about me. I don't have a real opinion about that and I think the existence of both is fine. We probably should clarify the documentation and say that they do the same things.
